### PR TITLE
fix #123 fix #125

### LINKED ISSUE #123: [EN] SQL exception during update
app version before update: v1.2.5
app version to update to: v2.0.1

SQL exception:
`An exception occurred while executing a query: SQLSTATE[42601]: Syntax error: 7 ERROR: syntax error at or near "CHANGE" LINE 1: ALTER TABLE "oc_gestion_client" CHANGE "siret" "legal_one" L... ^`

System info:
* NC version 23.0.0
* DB type: PostgreSQL v10.15
* OS: Ubuntu 20.04 LTS
* Webserver: Apache 2.4.41

### PR DESCRIPTION
**Pull request by baimard**
fix migration issue https://github.com/baimard/gestion/issues/123
fix baseurl issue #125 

Translation !

### CODE REVIEW

File: lib/Migration/Version20002Date20220201085856.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		foreach ($this->rows as $row) {
+     		$stmt = $this->db->prepare("UPDATE `".'*PREFIX*'."gestion_client` SET `legal_one` = ? WHERE id = ?;");
+        	$stmt->execute(array($row['siret'], $row['id']));
```
Comment (vitormattos) [inline lib/Migration/Version20002Date20220201085856.php:84]: You can use queryBuilder to don't get problem with different databases:

```suggestion
			$qb = $this->db->getQueryBuilder();
			$qb
				->update('gestion_client')
				->set('legal_one', $qb->createNamedParameter($row['siret']))
				->where($qb->expr()->eq('id', $qb->createNamedParameter($row['id'])))
				->execute();
```

File: lib/Migration/Version7Date20220125174818.php [hunk 2]
```diff
@@ -54,11 +54,11 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
...
-        if (!$table->hasColumn('siret')) {
-        }else{
-            $stmt = $this->db->prepare("ALTER TABLE `".'*PREFIX*'."gestion_client` CHANGE `siret` `legal_one` LONGTEXT");
-            $stmt->execute();
-        }
+        // if (!$table->hasColumn('siret')) {
+        // }else{
+        //     $stmt = $this->db->prepare("ALTER TABLE `".'*PREFIX*'."gestion_client` CHANGE `siret` `legal_one` LONGTEXT");
+        //     $stmt->execute();
+        // }
```
Comment (vitormattos) [inline lib/Migration/Version7Date20220125174818.php:61]: I don't like commented code, git already store the change log history
```suggestion
```

File: lib/Migration/Version20002Date20220201085856.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+use OCP\Migration\SimpleMigrationStep;
+use OCP\IDBConnection;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version20002Date20220201085856 extends SimpleMigrationStep {
+	/**
+	 * IDB
+	 */
```
Comment (vitormattos) [inline lib/Migration/Version20002Date20220201085856.php:19]: Unecessary
```suggestion
```

File: lib/Migration/Version20002Date20220201085856.php (hunk 1 above)
Comment (baimard) [reply lib/Migration/Version20002Date20220201085856.php:84]: I don't know what that do : 
(it's why I don't use) Could you explain ?

```
$qb->createNamedParameter($row['siret'])

Comment (vitormattos) [reply lib/Migration/Version20002Date20220201085856.php:78]: Is the QueryBuilder:
https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/query-builder.html

You will generate the query using PHP methods and when the `execute` method was executed, inside of `execute` all parts of query will be converted to SQL syntax compatible with current database.

Example.... If you need use `limit` on a query, Oracle have an specific syntax for limit. You don't need know this, you only will call the method `limit` and the QueryBuilder will do all rest of magic.

The `createNamedParameter` is to pass a parameter to the query. Is good run queries using parameters to optimize the execution plan of current database. Like that you already did on first code passing the parameters to method execute and using `?` to define the place of parameter. But, named parameter will put the parameter with name on query.

Comment (vitormattos) [reply lib/Migration/Version20002Date20220201085856.php:78]: After merge this PR, test the code. I only copied from other place and adapted to this necessity.

Comment (baimard) [reply lib/Migration/Version20002Date20220201085856.php:78]: Ok

**Review by baimard — COMMENTED** [review]
fix #123 
fix #125

**Pull request merged**